### PR TITLE
TDL-7634: Update schemas with missing fields

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-7634: Update schemas with missing fields
- Added fields in the schema as per the doc in the ticket.
- Added more missing fields in the schema from the doc: [QuickBooks Fields.pdf](https://github.com/singer-io/tap-quickbooks/files/8348653/QuickBooks.Fields.pdf)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_quickbooks/schemas/time_activities.json [hunk 1]
```diff
@@ -98,6 +98,57 @@
         "integer"
       ]
     },
+    "BreakHours": {
```
Comment (harshpatel4crest) [inline tap_quickbooks/schemas/time_activities.json:101]: As per the [doc](https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/timeactivity), BreakHours and BreakMinutes fields are added in a single value as `BreakHours BreakMinutes`. But in the API response, both fields are replicated as different fields.
image
Hence, segregated these fields in the schema.

File: tap_quickbooks/discover.py [hunk 2]
```diff
@@ -29,6 +31,18 @@ def _load_schemas():
 
     return schemas
 
+def _load_shared_schema_refs():
```
Comment (dbshah1212) [inline tap_quickbooks/discover.py:34]: Why did we remove the older load_schema and added a new one? Do add function comments with details.

File: tap_quickbooks/schemas/bills.json [hunk 3]
```diff
@@ -229,13 +210,125 @@
...
+          "$ref": "shared/ref_schema.json"
+        },
+        "TotalTax": {
+          "format": "singer.decimal",
+          "type": [
+            "null",
+            "string"
+          ]
+        },
+        "TaxLine": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/bills.json:255]: I can see `TaxLine` and `TaxLineDetail` are used at multiple places if all are the same then can we move these 2 in the shared folder?

File: tap_quickbooks/schemas/deposits.json [hunk 4]
```diff
@@ -142,6 +143,38 @@
               "null",
               "array"
             ]
+          },
+          "CustomField": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/deposits.json:147]: Can we use custom_field.json ref here?

File: tap_quickbooks/schemas/bills.json [hunk 5]
```diff
@@ -229,13 +210,125 @@
...
+    "TransactionLocationType": {
+      "type": [
+        "null",
+        "string"
+      ]
+    },
+    "RecurDataRef": {
+      "$ref": "shared/ref_schema.json"
+    },
+    "TxnTaxDetail": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/bills.json:243]: Can we use shared/txn_tax_detail.json ref here?

File: tap_quickbooks/schemas/deposits.json [hunk 6]
```diff
@@ -217,6 +250,124 @@
...
+        "string"
+      ]
+    },
+    "TxnSource": {
+      "type": [
+        "null",
+        "string"
+      ]
+    },
+    "TxnTaxDetail": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/deposits.json:282]: Can we use shared/txn_tax_detail.json ref here?

File: tap_quickbooks/schemas/purchases.json [hunk 7]
```diff
@@ -274,6 +145,45 @@
               "null",
               "string"
             ]
+          },
+          "LinkedTxn": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/purchases.json:149]: Can we move this into the shared folder and use ref here as I can see many appearances of it.

File: tap_quickbooks/discover.py (hunk 2 above)
Comment (harshpatel4crest) [reply tap_quickbooks/discover.py:34]: Initially, there were only 2 schema refs, hence, it was handled manually in the `do_discover` function. Now as, we have some reference schemas, thus, created a separate function to prepare schema refs.

File: tap_quickbooks/schemas/purchases.json (hunk 7 above)
Comment (harshpatel4crest) [reply tap_quickbooks/schemas/purchases.json:149]: Updated the code with a separate schema file for the `LinkedTxn` field with ref.

File: tap_quickbooks/schemas/deposits.json (hunk 6 above)
Comment (harshpatel4crest) [reply tap_quickbooks/schemas/deposits.json:224]: Updated.

File: tap_quickbooks/schemas/bills.json (hunk 5 above)
Comment (harshpatel4crest) [reply tap_quickbooks/schemas/bills.json:185]: Updated.

File: tap_quickbooks/schemas/deposits.json (hunk 4 above)
Comment (harshpatel4crest) [reply tap_quickbooks/schemas/deposits.json:118]: Yes, updated.

**Review by dbshah1212 — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Review by RushT007 — APPROVED** [review]

**Pull request merged**